### DISCUSSION TITLE
Fastest way to fill a large amount of individual pixels?

### DISCUSSION THREAD

**Discussion by BenevolentCheese** [ImageMagick/ImageMagick#5042] [Help]
I have a large set of precalculated pixel values of many colors that I need to write to an image. What's the most performant way to do that?

Using only a single color, I tested a single command with 10k `-draw point`s and it ran reasonably fast. I tested 10k commands with individual draw points and it was, unsurprisingly, extremely slow.

The first seems good, but that only uses a single color. I will have hundreds of colors. I could batch them into one command per color, but that would still leave me with hundreds of commands. Is there a good way to set all pixels with reasonable performance? If not in IM, can anyone recommend a library that would do this better?

**Comment by fmw42** [comment]
What exactly are you trying to do? Please show your code and identify your IM version and platform.

Are you trying to insert single pixel points of color into some image?

**Reply by BenevolentCheese** [reply]
I have a list of 1024x1024 RGB values and I need to set individual pixels in the image with those RGB values. I am on the newest version, on Windows.

**Reply by fmw42** [reply]
What version of ImageMagick?   What do you mean by 1024x1024 RGB values?  Do you mean you have 1024x1024=1048576 individual pixel values with color that you want to put into some image or that you have a 1024x1024 image that you want to overlay on another image?

**Comment by snibgo** [comment]
> I have a large set of precalculated pixel values  ...

What is the format of that data? If it is decimal text, with one pixel per line (or some other definition), channel values separated by a comma (or some other character), reading the data as "ftxt:" will be fairly fast.

I haven't done tests, but one of the binary formats should be faster, if your data is already binary.

But if you have only 10k or a million pixels to convert only once, performance shouldn't be concern.
